homeassistant/components/template/switch.py

```diff
@@ -134,22 +200,19 @@ class SwitchTemplate(TemplateEntity, SwitchEntity, RestoreEntity):
     def __init__(
         self,
         hass: HomeAssistant,
-        object_id,
         config: ConfigType,
-        unique_id,
+        unique_id: str | None,
     ) -> None:
         """Initialize the Template switch."""
-        super().__init__(
-            hass, config=config, fallback_name=object_id, unique_id=unique_id
-        )
-        if object_id is not None:
+        super().__init__(hass, config=config, fallback_name=None, unique_id=unique_id)
```

Comment:
Don't we need to still pass the fallback_name, or is it handled by this when rewriting the schema?
```
        if CONF_NAME not in entity_conf:
            entity_conf[CONF_NAME] = template.Template(object_id, hass)
```

Reply:
> Don't we need to still pass the fallback_name, or is it handled by this when rewriting the schema?
> 
> ```
>         if CONF_NAME not in entity_conf:
>             entity_conf[CONF_NAME] = template.Template(object_id, hass)
> ```

It's handled after in the object_id for loop.  This is how other template entities currently do it.  So essentially, it overwrites the entity_id if the object_id exists in the schema.

The other option is for me to change it back to the way it was, which was the object_id setting the fallback name.  But that means I need to get it first... which means the forloop before the super().

Reply:
OK